src/modules/dashboard/components/Vesting/VestingPageLayout.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { MessageDescriptor, defineMessage } from 'react-intl';
+import Button from '~core/Button';
+
+import Numeral from '~core/Numeral';
+import { SpinnerLoader } from '~core/Preloaders';
+import { SimpleMessageValues } from '~types/index';
+import { getFormattedTokenValue } from '~utils/tokens';
+
+import styles from './Vesting.css';
+
+const MSG = defineMessage({
+  loading: {
+    id: 'dashboard.Vesting.VestingPageLayout.loading',
+    defaultMessage: 'Loading',
+  },
+});
+
+const displayName = 'dashboard.Vesting.VestingPageLayout';
+
+interface Props {
+  title: JSX.Element;
+  buttonText: MessageDescriptor | string;
+  buttonTextValues?: SimpleMessageValues;
+  tableValues: {
+    label: JSX.Element;
+    value: string;
+  }[];
+  tokenDecimals: number;
+  isLoading: boolean;
+}
+
+const VestingPageLayout = ({
+  title,
+  tableValues,
+  buttonText,
+  tokenDecimals,
+  buttonTextValues,
+  isLoading,
+}: Props) => {
+  return isLoading ? (
+    <div className={styles.loader}>
+      <SpinnerLoader
+        loadingText={MSG.loading}
+        appearance={{ theme: 'primary', size: 'massive' }}
+      />
+    </div>
+  ) : (
+    <div className={styles.main}>
+      {title}
+      <div className={styles.table}>
+        {tableValues.map(({ label, value }) => (
+          <div className={styles.item}>
+            <div className={styles.label}>{label}</div>
+            <div className={styles.value}>
+              <Numeral value={getFormattedTokenValue(value, tokenDecimals)} />
+            </div>
+          </div>
+        ))}
```

Comment:
Items in this list need a unique `key`

![Screenshot from 2021-12-08 16-46-56](https://user-images.githubusercontent.com/1193222/145229242-6a10cab6-797d-47b9-a8cc-566854919286.png)